subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/ResolutionErrorRenderer.java

```diff
@@ -79,17 +80,17 @@ private void handleOutOfDateLocks(final LockOutOfDateException cause) {
 
     private void handleConflict(final VersionConflictException conflict) {
         registerError(output -> {
-            output.text("Dependency resolution failed because of conflict(s) on the following module(s):");
+            List<Pair<List<? extends ModuleVersionIdentifier>, String>> conflicts = conflict.getConflicts();
+            String plural = conflicts.size() > 1 ? "s" : "";
```

Comment:
💭 Might be worth making this a method in `TextUtil`?  (Though this wouldn't handle more complex pluralization rules, you'd need some more fully featured lib for that.)

Reply:
I thought about that too. just didn't get to it and since it's used a few times already I think it's worth it.